### ISSUE TITLE
instances have no internet access

### ISSUE THREAD

**Issue by Macmee** [CanonicalLtd/multipass#888]
Fresh install of MacOS with multipass. I installed the default ubuntu 18 image in multipass. I have no network access. For example if I `ping google.com` it hangs forever. It manages to resolve the domain to an IP address though.

**Comment by xluffy** [comment]
@Macmee I think this is not a bug of `multipass`, Can you check `/etc/resolv.conf`, It points to DNS server to 127.0.0.53. It means you can `ping 8.8.8.8` but can't ping to a domain.

Ref: https://github.com/systemd/systemd/issues/10298

**Comment by xluffy** [comment]
Hmm, same issue for windows version https://github.com/CanonicalLtd/multipass/issues/572.

Btw, I have same issue on my mac osx. That is all info.

- My `/etc/resolv.conf`
```
nameserver 127.0.0.53
options edns0
````
- `systemd-resolve --status`
```
systemd-resolve --status
Global
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      d.f.ip6.arpa
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 2 (enp0s2)
      Current Scopes: DNS
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 192.168.64.1
```
- Dig to google.com
```
root@mp-ubuntu-18:~# dig +short +tcp google.com @127.0.0.53
172.217.20.110
root@mp-ubuntu-18:~# dig +short +tcp google.com @192.168.64.1
172.217.20.110
```
- netstat info
```
> netstat -antpd | grep 53
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      732/systemd-resolve
tcp        0      0 127.0.0.53:53           127.0.0.1:60375         TIME_WAIT   -
```
- My firewall is disabled.
```
> ufw status
Status: inactive
```
- Test UDP port
```
# nc -v -u -z -w 3 127.0.0.53 53
Connection to 127.0.0.53 53 port [udp/domain] succeeded!
```
- Log
```
root@mp-ubuntu-18:~# journalctl -u systemd-resolved
-- Logs begin at Mon 2019-07-08 13:53:33 +07, end at Mon 2019-07-08 13:57:04 +07. --
Jul 08 13:53:39 mp-ubuntu-18 systemd[1]: Starting Network Name Resolution...
Jul 08 13:53:39 mp-ubuntu-18 systemd-resolved[732]: Positive Trust Anchors:
Jul 08 13:53:39 mp-ubuntu-18 systemd-resolved[732]: . IN DS 19036 8 2 49aac11d7b6f6446702e54a1607371607a1a41855200fd2ce1cdde32f24e8fb5
Jul 08 13:53:39 mp-ubuntu-18 systemd-resolved[732]: . IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d
Jul 08 13:53:39 mp-ubuntu-18 systemd-resolved[732]: Negative trust anchors: 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.arpa 19.172.in-addr.arpa 20.172.in-addr.arpa 21.172.in-addr.arpa 22.172.in
Jul 08 13:53:39 mp-ubuntu-18 systemd-resolved[732]: Using system hostname 'mp-ubuntu-18'.
Jul 08 13:53:39 mp-ubuntu-18 systemd[1]: Started Network Name Resolution.
Jul 08 13:57:04 mp-ubuntu-18 systemd-resolved[732]: Server returned error NXDOMAIN, mitigating potential DNS violation DVE-2018-0001, retrying transaction with reduced feature level UDP.
```

**Comment by gerboland** [comment]
@Macmee hey, `ping` not working is not a reliable check for internet connectivity as firewalls will interfere. Could you try `sudo apt update` and see if it completes? Or `wget google.ie`.


I suspect in your case ICMP packets are being blocked by MacOS' firewall Stealth mode. If you go to System Preferences->Security & Privacy, select the Firewall tab and go into "Firewall Options" you should see something like this:

![Screenshot 2019-07-12 at 14 47 33](https://user-images.githubusercontent.com/632769/61132903-d7e54300-a4ab-11e9-886f-eabb05aa197c.png)

The "Stealth Mode" option will block return ICMP packets that ping uses. Disable (need to hit "Ok" before it takes effect) to fix ping in the VM. But it's a good thing to keep enabled IMO.

**Comment by Saviq** [comment]
Please report back if you still have troubles.

**Comment by konart** [comment]
Not the OP, but I have the same problem. macOS 10.14.6, firewall is off.

Can't ping by domain name or an IP. `sudo apt update` also fails with `connection timed out` error for all mirrors.

**Comment by Saviq** [comment]
Hi @konart I just published a [Troubleshooting networking on macOS](https://discourse.ubuntu.com/t/troubleshooting-networking-on-macos/12901) guide on our Discourse, would you please have a look through it and see if there's anything that helps?

If you can't get it going, please file a new issue here or reply to the above post.

That `NXDOMAIN` warning seems to be unimportant, as `systemd-resolved` then simply switches to a different mode.

**Comment by bhurlow** [comment]
I encountered this issue, found that it was a locally running dnsmasq on OS X via homebrew that was creating DNS conflicts

**Comment by juandanielpujols** [comment]
I manually set the DNS to google's now it is working

**Comment by giaphui** [comment]
@juandanielpujols how did you configure your DNS manually?

**Comment by giaphui** [comment]
@konart did you manage to get it working? I'm having the same problem, using Mac, after installing Forticlient. Now all my Multipass VMs instances cannot reach the internet.

**Comment by L422Y** [comment]
Edit `/etc/systemd/resolved.conf` and add a DNS and fallback DNS server:

```
[Resolve]
DNS=[your local DNS server]
FallbackDNS=8.8.8.8
```
